### DISCUSSION TITLE
Trigger a workflow on completion of another workflow with right branch

### DISCUSSION THREAD

**Discussion by jsinh** [community/community#26362] [Actions]
<p>I have a workflow that is set up to run/get triggered when a PR is merged successfully.</p>
<p>Snippet from workflow:</p>
<pre><code class="lang-auto">name: "On PR"

on:  
  pull_request:
    types:
      - closed
    branches:
      - develop
</code></pre>
<p>An example is I have a branch <code>test</code> where I made some changes and then created PR to merge <code>test</code>'s changes into <code>develop</code>.</p>
<p>This workflow works fine and triggers as expected. On the workflow - run list it shows <code>test</code> as the branch while running the workflow - not <code>develop</code>.</p>
<p>Now I have another which is set to trigger when the above workflow is completed.</p>
<p>Snippet from this workflow:</p>
<pre><code class="lang-auto">on:
  workflow_run:
    workflows:
        - "On PR"
    branches:
        - develop
    types:
        - completed
</code></pre>
<p>I have two issues here.</p>
<ol>
<li>
<p>The Second workflow never gets triggered because the branches are set to <code>develop</code> if I set it to <code>test</code> then it does but branch <code>test</code> is dynamic and I have no control of what that would be so I want to trigger this considering the destination PR merge branch name for it.</p>
</li>
<li>
<p>This second workflow gets triggered even if my previous PR merge workflow fails - how can I set it to only get triggered if “On PR” is completed successfully and not on fail.</p>
</li>
</ol>
<p>Thanks in advance.</p>

**Comment by Simran-B** [comment, marked as answer]
<ol start="2">
<li>
<p>See <a href="https://github.community/t/workflow-run-completed-event-triggered-by-failed-workflow/128001/2" class="inline-onebox">Workflow_run completed event triggered by failed workflow</a></p>
<p>You can add <code>github.event.workflow_run.conclusion == 'success'</code> as <code>if</code> condition to jobs</p>
</li>
</ol>

**Comment by jsinh** [comment]
<p><a class="mention" href="/simran-b">@Simran-B</a> thanks for the solution on point 2. That worked like a charm!</p>
